lib/attr_json/record.rb

```diff
@@ -74,9 +74,12 @@ def attr_json(name, type, **options)
         # only if it hasn't already been done. WARNING we are using internal
         # Rails API here, but only way to do this lazily, which I thought was
         # worth it. On the other hand, I think .attribute is idempotent, maybe we don't need it...
+        #
+        # We set default to empty hash, because that 'tricks' AR into knowing any
+        # application of defaults is a change that needs to be saved.
         unless attributes_to_define_after_schema_loads[container_attribute.to_s] &&
                attributes_to_define_after_schema_loads[container_attribute.to_s].first.is_a?(AttrJson::Type::ContainerAttribute)
-            attribute container_attribute.to_sym, AttrJson::Type::ContainerAttribute.new(self, container_attribute)
+            attribute container_attribute.to_sym, AttrJson::Type::ContainerAttribute.new(self, container_attribute), default: -> {}
```

Comment:
Isn't `default: -> {}` saying the default is an empty `Proc`? I would have expected to see either:

```ruby
default: {} # the same hash literal is used to default all of them. Maybe that's OK?

# or..

default: -> { Hash.new } # a new hash is used each time

# which could also be written as

default: -> { {} }
```

Reply:
Yes, that's a typo, I could have sworn I wrote `default: -> { {} }`... yet tests pass anyway somehow very strange. (They def do not with no `default` here at all as pre-change). 

I think setting _anything_ here, including i'd guess, kinda oddly `default: nil`, might get AR to do what's needed.  

But I will change it what I intended `default: -> { {} }`. I trust the tests.